### PR TITLE
Issue 246 signup authentication and validation, save to database

### PR DESCRIPTION
**Pull request by almazhankapan**

### CODE REVIEW

File: back-end/routes/auth.js [hunk 1]
```diff
@@ -0,0 +1,76 @@
...
+  //hash password
+  const salt= await bcrypt.genSalt(10)
+  const hashedPassword = await bcrypt.hash(req.body.passwordHash, salt)
+
+  //create user object
+  const user = new User({
+    id: req.body.id, 
+    username: req.body.username, 
+    email: req.body.email, 
+    passwordHash: hashedPassword
```
Comment (sam-eng) [inline back-end/routes/auth.js:61]: I sent you some comments on Discord, but I've been looking over this and the new user created should also have a `bio`, `img`, and `shows` field (as per the User Model specifications)

**Review by benrkaplan — APPROVED** [review]

File: back-end/routes/auth.js [hunk 2]
```diff
@@ -0,0 +1,123 @@
...
+const app = require('express').Router()
+const User=require('../models/User')
+const passport = require('passport');
+const localStrategy = require('passport-local').Strategy;
+
+const bcrypt = require('bcrypt') //encrypt password
+const { body, validationResult } = require('express-validator')
+const expressSession = require('express-session') 
+app.use(express.urlencoded({ extended: true }))
+app.use(expressSession({'secret':'any','saveUninitialized':false, 'resave':false}))
```
Comment (sam-eng) [inline back-end/routes/auth.js:11]: This is more a question than a change request, but does the code work if `expressSession` is removed? If so, then could you remove references to it?

File: back-end/test/getUser.js [hunk 3]
```diff
@@ -1,43 +1,48 @@
...
   let stub
 
   afterEach(() => {
     stub.restore()
   })
 
+//sinon.stub(modelName, 'findOne').returns({object to be returned})
   it('should return 200 OK and data for a valid id', async () => {
-    stub = sinon.stub(axios, 'get').resolves({
+    stub = sinon.stub(User, 'findOne').returns({
```
Comment (sam-eng) [inline back-end/test/getUser.js:22]: Were you able to get these tests working? Or are they commented out because there were issues?

**Review by sam-eng — APPROVED** [review]
I left a few comments, but other than those, it looks pretty good to me.

File: back-end/test/getUser.js (hunk 3 above)
Comment (almazhankapan) [reply back-end/test/getUser.js:22]: i probably will create another task for testing hehe; for now will revert to how it was probably

**Pull request merged**